src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs

```diff
@@ -57,18 +66,33 @@ public Uri SourceUri
             {
                 var requestTimeout = TimeSpan.FromSeconds(timeoutInSecond);
                 tokenSource.CancelAfter(requestTimeout);
-
+                bool packageUseTempApiKey = false;
+                bool symbolUseTempApiKey = false;
                 var apiKey = getApiKey(_source);
 
-                await PushPackage(packagePath, _source, apiKey, requestTimeout, log, tokenSource.Token);
+                // If user push to https://nuget.smbsrc.net/ directly, use temp api key.
+                if (_source.Equals(NuGetConstants.DefaultSymbolServerUrl, StringComparison.OrdinalIgnoreCase))
+                {
+                    packageUseTempApiKey = true;
+                }
+
+                await PushPackage(packagePath, _source, apiKey, requestTimeout, log, packageUseTempApiKey, tokenSource.Token);
 
                 // symbolSource is only set when:
                 // - The user specified it on the command line
                 // - The package source is nuget.org and the default symbol source is used
                 if (!string.IsNullOrEmpty(symbolSource))
                 {
                     string symbolApiKey = getSymbolApiKey(symbolSource);
-                    await PushSymbols(packagePath, symbolSource, symbolApiKey, requestTimeout, log, tokenSource.Token);
+
+                    // If symbol source is nuget.smbsr.net, always use temp key.
+                    // If package source is nuget.org and symbol api key is the same as package api key, use temp key
+                    if (symbolSource.Equals(NuGetConstants.DefaultSymbolServerUrl, StringComparison.OrdinalIgnoreCase)
+                        || (IsSourceNuGetOrg() && apiKey == symbolApiKey))
```

Comment:
Why does it matter if it is nuget.org? We discussed not special casing this I thought.